pkg/virtualKubelet/liqoNodeProvider/nodeProvider.go

```diff
@@ -58,11 +69,27 @@ func (p *LiqoNodeProvider) Ping(ctx context.Context) error {
 	start := time.Now()
 	klog.V(4).Infof("Checking whether the remote API server is ready")
 
-	_, err := p.remoteDiscoveryClient.RESTClient().Get().AbsPath("/livez").DoRaw(ctx)
+	// Get the foreigncluster using the given clusterID
+	clusterIDSelector := client.MatchingLabelsSelector{Selector: labels.Set{liqodiscovery.ClusterIDLabel: p.foreignClusterID}.AsSelector()}
+	unstr, err := p.dynClient.Resource(discoveryv1alpha1.ForeignClusterGroupVersionResource).
+		List(ctx, metav1.ListOptions{LabelSelector: clusterIDSelector.String()})
 	if err != nil {
-		klog.Errorf("API server readiness check failed: %v", err)
+		klog.Error("Failed to list foreignclusters: %v", err)
+		return err
+	}
+	var foreignClusters discoveryv1alpha1.ForeignClusterList
+	utilruntime.Must(runtime.DefaultUnstructuredConverter.FromUnstructured(unstr.UnstructuredContent(), &foreignClusters))
+	if len(foreignClusters.Items) == 0 {
+		err := errors.Errorf("Failed to get foreigncluster %q", p.foreignClusterID)
+		klog.Error(err)
 		return err
 	}
+	fc := &foreignClusters.Items[0] // there should be only one foreigncluster with the given clusterID
+
+	// Check the foreign API server status
+	if !foreignclusterutils.IsAPIServerReady(fc) {
+		return fmt.Errorf("API server readiness check failed")
```

Comment:
Can you write which cluster is causing the problem (by adding the clusterID and/or the clusterName in the log)?